src/Bigcommerce/Api/Resource.php

```diff
@@ -30,6 +30,11 @@ class Resource
 	 */
 	protected $ignoreIfZero = array();
 
+	/**
+	 * @var array
+	 */
+	protected $fieldMap = array();
```

Comment:
I cant comment around the line, but does this need `__unset` magic method to support line 71 / https://github.com/bigcommerce/bigcommerce-api-php/pull/115/files#diff-ee31cb3bb777ad8d525ec3bb5d7dbcc9R71

seems like its calling `unset` on the wrong thing, `$this->$field` instead of `$this->fields->$field`..unless this is how this works.
/me git clones the repo